### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by apt-itude**
Initial implementation of parser for Tilt Hydrometer devices to be used by Home Assistant integration.

This has been running successfully as part of a custom component on my Home Assistant for a few weeks now.

@Ernst79 Apologies: I let the invite to contribute directly to the upstream repo expire

### CODE REVIEW

File: src/tilt_ble/parser.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        self.update_sensor(
+            key=DeviceClass.TEMPERATURE,
+            device_class=DeviceClass.TEMPERATURE,
+            native_unit_of_measurement=Units.TEMP_FAHRENHEIT,
+            native_value=temperature,
+        )
+        self.update_sensor(
+            key=DEVICE_CLASS_SG,
+            device_class=DEVICE_CLASS_SG,
+            native_unit_of_measurement=UNITS_SG,
```
Comment (apt-itude) [inline src/tilt_ble/parser.py:88]: I wasn't sure what to do about device class and units for this sensor. The device class is relatively unusual (specific gravity) and it's really unit-less, but from what I can tell omitting units doesn't work so well in HA.

Do I need to define this in the `sensor-state-data` library first?

Comment (bdraco) [reply src/tilt_ble/parser.py:88]: Yes it needs something in sensor state data first

Comment (apt-itude) [reply src/tilt_ble/parser.py:88]: Thanks! Can it actually be unit-less, or do I just need to make something up? Like `G`?

Comment (bdraco) [reply src/tilt_ble/parser.py:88]: When you bring it into Home Assistant, map the units to `None`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          fetch-depth: 0
+
+      # Run semantic release:
+      # - Update CHANGELOG.md
+      # - Update version in code
+      # - Create git tag
+      # - Create GitHub release
+      # - Publish to PyPI
+      - name: Python Semantic Release
+        uses: relekang/python-semantic-release@v7.27.1
```
Comment (bdraco) [inline .github/workflows/ci.yml:85]:
```suggestion
        uses: relekang/python-semantic-release@v7.31.4
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          fetch-depth: 0
+      - uses: wagoid/commitlint-github-action@v4.1.11
+
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.7"
+          - "3.8"
```
Comment (bdraco) [inline .github/workflows/ci.yml:40]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          - "3.9"
+          - "3.10"
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macOS-latest
```
Comment (bdraco) [inline .github/workflows/ci.yml:46]:
```suggestion
```

**Comment by bdraco** [comment]
@apt-itude What is your PyPI handle?

**Comment by bdraco** [comment]
@apt-itude  I'm on holiday starting tomorrow so I didn't want to leave you hanging.

I got the auto publication workflow and tokens working so as long as the CI passes it should publish new changes based on the commit message format.  (https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)

@Ernst79 can add you to the PyPI package.

File: src/tilt_ble/parser.py (hunk 1 above)
Comment (Ernst79) [reply src/tilt_ble/parser.py:88]: I think the device class should also send None to HA, isn’t it? Specific gravity is not a HA device class.

**Comment by apt-itude** [comment]
> @apt-itude What is your PyPI handle?

In case you still need it, same as here: `apt-itude`

**Pull request merged**